lib/concurrent-ruby/concurrent/collection/ruby_timeout_queue.rb

```diff
@@ -0,0 +1,79 @@
+module Concurrent
+  module Collection
+    # @!visibility private
+    # @!macro ruby_timeout_queue
+    class RubyTimeoutQueue < ::Queue
+      def initialize(*args)
+        if RUBY_VERSION >= '3.2'
+          raise "#{self.class.name} is not needed on Ruby 3.2 or later, use ::Queue instead"
+        end
+
+        super(*args)
+
+        @mutex = Mutex.new
+        @cond_var = ConditionVariable.new
+      end
+
+      def push(obj)
+        @mutex.synchronize do
+          super(obj)
+          @cond_var.signal
+        end
+      end
+      alias_method :enq, :push
+      alias_method :<<, :push
+
+      def pop(non_block = false, timeout: nil)
+        if non_block && timeout
+          raise ArgumentError, "can't set a timeout if non_block is enabled"
+        end
+
+        if non_block
+          super(true)
+        elsif timeout
+          @mutex.synchronize do
+            while timeout.positive? # handle spurious wakeups
+              begin
+                return super(true)
+              rescue ThreadError
+                # empty, wait until woken up
+              end
+
+              timed_out, timeout = ns_timed_out?(timeout) { @cond_var.wait(@mutex, timeout) }
```

Comment:
Could you simplify to something like https://github.com/ruby/timeout/blob/607d8c6fbe4d86db1cf22846e89198b47cec7161/lib/timeout.rb#L105-L109 (which I wrote) and avoid the extra method?
That seems a lot simpler and I believe it's fully correct.

Reply:
Done.